### PR TITLE
Parallelize CSV processing in dc-import

### PR DESCRIPTION
**Pull request by pradh**
This is a primitive version where the underlying classes use coarser-grained locking, and might need improvements.

The following classes used during CSV processing are made thread-safe:
1. LogWrapper
2. ExistenceChecker
3. StatChecker
4. ExternalIdResolver
5. StatVarState

(1), (2), (3) use simple method-level "synchronized" primitive. (4) uses RW locks. (5) uses concurrent hashmap.

Used the census S1702 table import to measure performance (in units of M/m...aka Million nodes per minute) on my 8-core cloud-top, where 5 threads did best (anything more worsens throughput, probably due to increased contention).

Mode                                                                                         |   `--num-threads=1`  |  `--num-threads=5`
----------------------------------------------------------|----------------------|----------------------
`genmcf --existence-checks=False --resolution=LOCAL` |       1.1 M/m                 | 4.6 M/m
`lint --existence-checks=False`                                            |       1.7 M/m                 | 7.5 M/m
`genmcf --existence-checks=True --resolution=LOCAL`   |      0.8 M/m                 | 1.4 M/m
`lint --existence-checks=True`                                             |       1.2 M/m                 | 2 M/m

On my 2-core laptop throughput is halved (without existence-checks), or similar (with existence-checks).

Improvements are lower with existence-checks likely because a "batch flush" blocks all threads, and that probably needs optimizing.

Finally, the --num_threads=1 case does similar to the current head, so this change should be performance neutral by default.

### CODE REVIEW

**Review by chejennifer — APPROVED** [review]
Very cool!

**Comment by pradh** [comment]
Thanks for the review!

**Review by beets — COMMENTED** [review]
sorry for the late review - this is very cool!

File: tool/src/main/java/org/datacommons/tool/Processor.java [hunk 1]
```diff
@@ -186,103 +176,166 @@ private void processNodes(Mcf.McfType type)
...
+            + " rows, {} nodes)",
+        csvFile.getName(),
+        numRowsProcessed,
+        numNodesProcessed);
+    writerPair.close();
+  }
+
+  // Encloses a pair of writers for success and corresponding failure types, and creates the file
+  // on-demand when a write comes in.
+  class WriterPair {
```
[the remark this answers is not in the document]
Comment (pradh) [reply tool/src/main/java/org/datacommons/tool/Processor.java:288]: I will refactor in follow-on PR.  This class started out as a struct, but has enough logic now to move out...

Thanks for the review!

**Pull request merged**